### PR TITLE
Feature/omega 30 new han port info view

### PR DESCRIPTION
**Pull request by steinmy**
OMEGA-30

## What is the reason for this change?

HAN-port page has a new design

### What is the expected result?

HAN-port page should look the same as in the Figma sketches.

https://www.figma.com/file/sJHoScmQRq0hlr6AcHI9WB/Elvia.no?node-id=1%3A33

## How is it solved?

By changing the layout and styling of the HAN-port page and some related components.

## Any changes/updates that may affect others?

|                       | Changed |
| --------------------- | ------- |
| Unit tests            |        |
| Dependencies          |         |
| Environment variables |         |
| Build scripts         |         |

## How can it be tested and verified?

- [ ] Look at the han port page in multipe screen sizes
- [ ] Try to use the functionality of the HAN-port page, and see that it works and look correct in multipe screen sizes

### CODE REVIEW

File: packages/web/src/components/HANPortPhone/HANPortPhone.tsx [hunk 1]
```diff
@@ -67,6 +70,21 @@ export const HANPortPhone = ({
...
 
+  const createButton = (
+    onClick: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void,
+    iconName: string,
+    text: string
+  ) => (
+    <Button onClick={onClick} asType={ButtonType.Tertiary} size="small">
+      <ElvisIcon
+        iconName={iconName}
+        fontSize={16 * (16 / 14)}
```
Comment (steinmy) [inline packages/web/src/components/HANPortPhone/HANPortPhone.tsx:81]: Is there a prettier way to do this? I found that this was needed in order to get the icon to be 16 px wide due to relative font sizes.

File: packages/web/src/components/HANPortList/HANPortList.tsx [hunk 2]
```diff
@@ -52,41 +69,71 @@ export const HANPortList = ({
...
+        createLabel(status),
+        createCheckbox(status, contractId, index),
+      ])
   )
+  const tableData = headers.concat(tableRows)
 
   return (
-    <div className={css.list}>
-      {hanPortEventList.length > 1 ? (
-        <Expandable
```
Comment (steinmy) [inline packages/web/src/components/HANPortList/HANPortList.tsx:81]: The `Expandable` component is now not used anywhere. Should it be deleted?

**Comment by ragnhildlarsen** [comment]
Have you added any route for this so it can be tested?

**Comment by marasva** [comment]
https://5f68bce23920b81925704416--elvia-test.netlify.app/minside/hanport/4c9923d4-e83f-48cd-b008-04cb879e8ed9. Here is a link @ragnhildlarsen

**Comment by marasva** [comment]
image
The HAN-port is supposed to be a part of the tabs? In that case change / add the routes in CustomerDetail.tsx and minside.tsx? Then the "Tilbake til målersiden"-button should also be changed / removed.

**Comment by marasva** [comment]
image 

Don't know if it is anything you should do about it, but on mobile (and maybe normal screen on desktop) the warning box shows up on top of the screen, so it is not possible to see what is actually wrong. Maybe it should popup right above the "Bestill åpning" button, so it is possible to see what is actually wrong? 

Or that it is not even possible to tap the button (greyed out) if the requirements are not satisfied. But this last point is probably out of scope for this task?

**Comment by marasva** [comment]
image

The warning icon is not centered. But this should maybe be centered generally and not just for HAN-port?

**Comment by marasva** [comment]
image
On mobile if you uncheck the checkbox the box is still colored green even though it should be without color. To remove the color you have to actively tap somewhere else on the screen. Can be perceived as a bit misleading.

**Comment by steinmy** [comment]
> Don't know if it is anything you should do about it, but on mobile (and maybe normal screen on desktop) the warning box shows up on top of the screen, so it is not possible to see what is actually wrong. Maybe it should popup right above the "Bestill åpning" button, so it is possible to see what is actually wrong?
> 
> Or that it is not even possible to tap the button (greyed out) if the requirements are not satisfied. But this last point is probably out of scope for this task?

Nice catch! This should definitely be looked into, but perhaps as another task or at least a separate pull request. I have informed Malin, so we'll see what she says.

**Comment by steinmy** [comment]
@marasva 
> 
> image
> 
> The warning icon is not centered. But this should maybe be centered generally and not just for HAN-port?

Thanks for the feedback! I have noticed this, and reported it as a bug. It is however also a problem in the old solution, so I don't think it needs to be addressed in this pull request, but rather as a separate task about fixing the warning component in general.

**Comment by steinmy** [comment]
@marasva 
> The HAN-port is supposed to be a part of the tabs? In that case change / add the routes in CustomerDetail.tsx and minside.tsx? Then the "Tilbake til målersiden"-button should also be changed / removed.

I was going to do this in a separate pull request after this one is done, but I'm looking into it now, and if it turns out to be quickly done I can include it in this one.

File: packages/web/src/components/HANPortList/HANPortList.tsx [hunk 3]
```diff
@@ -52,41 +69,71 @@ export const HANPortList = ({
...
-        >
-          {hanPortEventList}
-        </Expandable>
-      ) : (
-        hanPortEventList
-      )}
-    </div>
+    <Table
+      data={tableData}
+      className={classnames(className, css.hanPortList)}
```
Comment (ragnhildlarsen) [inline packages/web/src/components/HANPortList/HANPortList.tsx:136]: The css class `hanPortList` seems to not exist anywhere. Should it be removed?

File: packages/web/src/templates/HANPort/HANPort.tsx [hunk 4]
```diff
@@ -202,92 +183,84 @@ export const HANPort: React.FC<PageQueryResponse> = ({
...
-              data={contractList}
+      <section className={css.infoTextWrapper}>
+        <div>{terms.asideContentText}</div>
+        <div className={css.illustration}>
+          <HddHanPortColor />
+        </div>
+      </section>
+      <section className={css.form}>
+        <TitleMedium className={css.title}>
+          {hasClosedStausOrIsEmpty
```
Comment (ragnhildlarsen) [inline packages/web/src/templates/HANPort/HANPort.tsx:194]: typo in `hasClosedStausOrIsEmpty`, should probably be `hasClosedStatusOrIsEmpty`

File: packages/web/src/components/HANPortList/HANPortList.tsx (hunk 3 above)
Comment (steinmy) [reply packages/web/src/components/HANPortList/HANPortList.tsx:136]: Ah, it was removed in a recent refactoring. Thanks for the heads up!

File: packages/web/src/templates/HANPort/HANPort.tsx [hunk 5]
```diff
@@ -119,28 +115,13 @@ export const HANPort: React.FC<PageQueryResponse> = ({
     goToMeterpointLink: getTerm(Terms.minSideHanOrderBackToMeterpointLink),
   }
 
-  const labelType = cond([
-    [equals(HANPortEventStatus.OPEN), always(LabelType.Safe)],
-    [equals(HANPortEventStatus.OPENING_REQUESTED), always(LabelType.Warning)],
-    [T, always(LabelType.Default)],
-  ])
-
   const contractList = sortedContractsAndHANEvents.map(
```
Comment (ragnhildlarsen) [inline packages/web/src/templates/HANPort/HANPort.tsx:118]: Ensure that the contracts shown are connected to the customer relationship that is selected. E.g. Per Hansen only have 2 contracts related to his customer relationship and Testbedrift 1 has 4 contracts. You could have a look at Erik's PR for meterpoints table to see how he only retrieves the contracts relevant to the customer relationship that is selected.

File: packages/web/src/templates/HANPort/HANPort.tsx [hunk 6]
```diff
@@ -202,92 +183,84 @@ export const HANPort: React.FC<PageQueryResponse> = ({
...
-                    []
-                  )(formErrorState)}
-                </ul>
-              </AlertMessage>
-            )}
-            <HANPortList
-              name={hanPortListName}
-              data={contractList}
+      <section className={css.infoTextWrapper}>
+        <div>{terms.asideContentText}</div>
```
Comment (ragnhildlarsen) [inline packages/web/src/templates/HANPort/HANPort.tsx:187]: typo in `asideContentText`?

**Review by ragnhildlarsen — CHANGES_REQUESTED** [review]
Had a few comments but overall it looks really good! :)

File: packages/web/src/templates/HANPort/HANPort.tsx (hunk 6 above)
Comment (steinmy) [reply packages/web/src/templates/HANPort/HANPort.tsx:187]: resolved after discussion, not a typo

File: packages/web/src/components/HANPortPhone/HANPortPhone.tsx (hunk 1 above)
Comment (steinmy) [reply packages/web/src/components/HANPortPhone/HANPortPhone.tsx]: Fixed

File: packages/web/src/templates/HANPort/HANPort.tsx (hunk 5 above)
Comment (steinmy) [reply packages/web/src/templates/HANPort/HANPort.tsx:118]: Postponed to next PR after discussion

**Comment by ragnhildlarsen** [comment]
Two comments on mobile view:
- Text next to edit icon should be left aligned aligned according to the sketches: https://www.figma.com/file/sJHoScmQRq0hlr6AcHI9WB/Elvia.no?node-id=10172%3A921
- it also looks like the text should be "Rediger"?
- Looks like the checkbox should be aligned at the top with the text next to it? 

Screenshot 2020-09-26 at 14 47 27

File: packages/web/src/components/HANPortList/HANPortList.tsx [hunk 7]
```diff
@@ -1,22 +1,39 @@
 import React, { useState, useEffect } from 'react'
-import { InputOption, Expandable, reduceTree } from '@elvia/ui'
+import { InputOption } from '@elvia/ui/src/components/InputOption/InputOption'
+import { reduceTree } from '@elvia/ui/src/utils/object'
+import { Label, LabelType } from '@elvia/ui/src/components/Label/Label'
 import { getTerm, Terms } from '@utils/getTerm'
 import css from './HANPortList.module.scss'
+import classnames from 'classnames'
```
Comment (ragnhildlarsen) [inline packages/web/src/components/HANPortList/HANPortList.tsx:7]: classnames import is unused

File: packages/web/src/components/HANPortPhone/HANPortPhone.tsx [hunk 8]
```diff
@@ -84,28 +102,35 @@ export const HANPortPhone = ({
...
+        </div>
+        <div>
+          <TextCaps className={css.phoneInfoTitle}>{terms.header}</TextCaps>
+          <TextDescription>
+            {currentPhoneNumber ? (
+              <>{terms.sendToPhoneText}</>
+            ) : (
+              terms.missingPhoneNumberText
+            )}
+          </TextDescription>
```
Comment (ragnhildlarsen) [inline packages/web/src/components/HANPortPhone/HANPortPhone.tsx:117]: Why is the first term wrapped in `<></>` and not the second one after `:`?

File: packages/web/src/templates/HANPort/HANPort.module.scss [hunk 9]
```diff
@@ -1,96 +1,79 @@
 @import '../../theme';
 
 .title {
+  font-size: 24px;
```
Comment (ragnhildlarsen) [inline packages/web/src/templates/HANPort/HANPort.module.scss:4]: emCalc?

File: packages/web/src/templates/HANPort/HANPort.module.scss [hunk 10]
```diff
@@ -1,96 +1,79 @@
...
 
-.wrapper {
-  display: flex;
-  flex-direction: column;
-
-  @media #{$tablet} {
-    align-items: flex-start;
-    flex-direction: row;
+  @media #{$mobileOnly} {
+    font-size: 20px;
```
Comment (ragnhildlarsen) [inline packages/web/src/templates/HANPort/HANPort.module.scss:10]: emCalc?

File: packages/web/src/templates/HANPort/HANPort.module.scss [hunk 11]
```diff
@@ -1,96 +1,79 @@
...
-
-  @media #{$tablet} {
-    max-width: 244px;
-  }
+  flex-direction: row;
+  margin-right: spaceCalc(5);
+
+  div {
+    p {
+      font-size: 20px;
```
Comment (ragnhildlarsen) [inline packages/web/src/templates/HANPort/HANPort.module.scss:32]: emCalc?

File: packages/web/src/templates/HANPort/HANPort.module.scss [hunk 12]
```diff
@@ -1,96 +1,79 @@
...
+      line-height: 160%;
+      margin-bottom: spaceCalc(4);
+
+      @media #{$mobileOnly} {
+        font-size: 18px;
```
Comment (ragnhildlarsen) [inline packages/web/src/templates/HANPort/HANPort.module.scss:37]: emCalc?

**Comment by steinmy** [comment]
@ragnhildlarsen 
> 
>     * it also looks like the text should be "Rediger"?
> 

This is controlled in Contentful, and I will change the text when the code is in production

File: packages/web/src/components/HANPortPhone/HANPortPhone.tsx (hunk 8 above)
Comment (steinmy) [reply packages/web/src/components/HANPortPhone/HANPortPhone.tsx:117]: Nice catch! This is a leftover from the old code and can be deleted.

**Comment by t0ff3r** [comment]
@steinmy whats the status here? Is it ready for a re-review from @ragnhildlarsen ?

**Comment by steinmy** [comment]
> 
> 
> @steinmy whats the status here? Is it ready for a re-review from @ragnhildlarsen ?

Yes, she's on it :)

**Review by ragnhildlarsen — APPROVED** [review]

**Comment by eTallang** [comment]
I guess that the HAN-port tab is introduced in another task?

**Pull request merged**